pyhms/sprout/sprout_filters.py

```diff
@@ -89,16 +84,13 @@ def __init__(self, limit: int) -> None:
 
     def __call__(
         self,
-        candidates: Dict[AbstractDeme, Tuple[Dict[str, float], List[Individual]]],
+        candidates: dict[AbstractDeme, DemeCandidates],
         _,
-    ) -> Dict[AbstractDeme, Tuple[Dict[str, float], List[Individual]]]:
+    ) -> dict[AbstractDeme, DemeCandidates]:
         for deme in candidates.keys():
-            candidates[deme][1].sort(key=lambda ind: ind.fitness)
-            if len(candidates[deme][1]) > self.limit:
-                candidates[deme] = (
-                    candidates[deme][0],
-                    candidates[deme][1][: self.limit],
-                )
+            if len(candidates[deme].individuals) > self.limit:
+                candidates[deme].individuals.sort(key=lambda ind: ind.fitness)
```

Comment:
I used this approach previously, but it takes 3 lines of code instead of one. By difficulty in tracking the internal state you mean a perspective of someone reading the code or some algorithmic quality?

Reply:
One liner could look like this:
```
candidates[deme].individuals = sorted(candidates[deme].individuals, key=lambda ind: ind.fitness)[: self.limit]
```

Reply:
Ok, I added the temporary variable for two filter classes, where the issue you describe could be the most problematic.